File: Code/Greedy/Largest_Number_179.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    string largestNumber(vector<int> n)
    {
        vector<string> nums;
        int countZero = 0;
        for (int i : n)
        {
            if (i == 0)
                ++countZero;
            nums.push_back(to_string(i));
        }
        if (countZero == n.size())
            return "0";
        sort(nums.begin(), nums.end(), [](string &a, string &b)
             { return (a + b) > (b + a); });
        string ans;
        for (auto i : nums)
            ans += i;
        return ans;
    }
};
int main()
{
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif
    Solution obj;
    auto ans = obj.largestNumber({1, 8, 6, 2, 5, 4, 8, 3, 7});
    // cout << ans;
    return 0;
}

File: java/Greedy_java/Largest_Number_179.java
package java.Greedy_java;

import java.util.*;
import java.io.*;

// import static java.lang.Math.max;
// import static java.lang.Math.min;
// import static java.lang.Math.abs;
// import static java.lang.System.out;

public class Largest_Number_179 {

    public String largestNumber(int[] n) {
        String nums[] = new String[n.length];

        for (int i = 0; i < n.length; ++i) {
            nums[i] = n[i] + "";
        }
        Arrays.sort(nums, new Comparator<String>() {
            @Override
            public int compare(String i, String j) {
                String s1 = i + j;
                String s2 = j + i;
                return s2.compareTo(s1);
            }
        });

        if (nums[0].charAt(0) == '0')
            return "0";

        String ans = "";
        for (String i : nums)
            ans += i;
        return ans;
    }

    public static void main(String[] args) throws FileNotFoundException {
        File f = new File("output.txt");

        // Scanner ip = new Scanner(System.in);
        try (Scanner ip = new Scanner(new File("input.txt"));
                PrintWriter op = new PrintWriter(f);) {
            Largest_Number_179 obj = new Largest_Number_179();
            int grid[] = { 1, 8, 6, 2, 5, 4, 8, 3, 7 };
            op.println(obj.largestNumber(grid));

        }
    }

}
